src/types.ts

```diff
@@ -49,6 +49,15 @@ export interface DirectusImageProps extends Omit<DirectusAssetProps, 'download'
   quality?: number;
   /** The fit of the thumbnail while always preserving the aspect ratio. */
   fit?: 'cover' | 'contain' | 'inside' | 'outside';
+  /** What file format to return the image in. */
+  format?: 'auto' | 'jpg' | 'png' | 'webp' | 'tiff';
+  /** Disable image up-scaling. */
+  withoutEnlargement?: boolean;
+  /** An array of sharp operations to apply to the image. */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  transforms?: [string, ...any[]][];
+  /** Key for Storage Asset Preset ( https://docs.directus.io/user-guide/cloud/project-settings.html#files-thumbnails ). */
+  key?: string;
```

Comment:
Can we express the fact that when key is used, all other options must not specified?

Reply:
I suggest adding a TSDoc comment for this configuration, even though it's technically mentioned in the Directus documentation. If you're okay with including TSDoc comments, it will enhance the codebase's documentation and make it more accessible for developers.

Reply:
I was talking about forcing it using typescript alone

Reply:
That would be great! However, I'm not quite familiar with how to implement that. If it's possible, it would definitely be the ideal option.

Reply:
I added an interface overload. Was that what you ment?

Reply:
I mean that this are valid:

`<DirectusImage asset='11263b49-13b9-4ed8-ba39-01fe7cbeb284' key="key" />`

Even this, without any prop sohuld be (as per documentation):

`<DirectusImage asset='11263b49-13b9-4ed8-ba39-01fe7cbeb284'  />`

But the following would not (`key` and `custom` props togheter):

`<DirectusImage asset='11263b49-13b9-4ed8-ba39-01fe7cbeb284' key="key" width={500} height={500} />`

I'm not a TypeScript guru so I can't really express this constraint but I'm sure it's possibile.

Reply:
Success! It took me a while, but I finally figured it out. The implementation is working flawlessly, and now all that's left is to tidy up the code and commit it.

I spent nearly an hour trying to understand why the `key` wasn't being passed as a prop. 🤦